track.py

```diff
@@ -61,12 +61,40 @@ class Track:
     @staticmethod
     def from_xml_file(filename):
         xmldoc = parse(filename)
+        width = get_track_root_info(xmldoc, 'width')
+        default_profil_steps_length = get_track_root_info(xmldoc, 'profil steps length')
+
         segments = []
         for section in get_main_track(xmldoc).childNodes:
             if section.nodeType == Node.ELEMENT_NODE:
-                segments.append(Segment.from_xml(section))
+                s = Segment.from_xml(section)
+                if s.end_radius is None:
+                    segments.append(s)
+                else:
+                    profil_steps_length = default_profil_steps_length
+                    if s.profil_steps_length is not None:
+                        profil_steps_length = s.profil_steps_length
+                    length = abs((s.radius + s.end_radius)/2.0 * s.arc)
+                    num_steps = int(length/profil_steps_length) + 1
+                    if num_steps == 1:
+                        segments.append(s)
+                    else:
+                        dlen = length/float(num_steps - 1)
```

Comment:
Oh, well, see [track4.cpp#L1442](https://github.com/robotika/Roborace-Dreams/blob/56a8771d1ab385c4c81677ee66a46686749b3135/src/modules/track/trackv1/track4.cpp#L1442) and [track4.cpp#L1446](https://github.com/robotika/Roborace-Dreams/blob/56a8771d1ab385c4c81677ee66a46686749b3135/src/modules/track/trackv1/track4.cpp#L1446). That is why I didn't want to reimplement it :frowning_face: .